### PR TITLE
[NIL-125] Add tech spec and quality stmt to Indicator

### PR DESCRIPTION
**Pull request by mchalejeiwc**
Add two new fields to the Indicator document type. Amalgamate these in the RTL within the Resources section. Also fix the empty-html content issue in the RTL.

### CODE REVIEW

File: repository-data/webfiles/src/main/resources/site/freemarker/nationalindicatorlibrary/indicator.ftl [hunk 1]
```diff
@@ -115,19 +115,32 @@
...
-                                <a title="${attachment.text}" href="<@hst.link hippobean=attachment.resource/>" onClick="logGoogleAnalyticsEvent('Download attachment','Indicator','${attachment.resource.filename}');">${attachment.text}</a>
-                                <span class="fileSize">(<@formatFileSize bytesCount=attachment.resource.length/>)</span>
-                            </li>
-                        </#list>
-                    </ul>
-                
+                        <ul data-uipath="nil.indicator.resources">
+                            <#if indicator.details.qualityStatementUrl?has_content>
+                                <li class="attachment">
+                                    <a href="${indicator.details.qualityStatementUrl}" target="_blank"><@fmt.message key="headers.qualityStatement"/></a>
```
Comment (jacsamell) [inline repository-data/webfiles/src/main/resources/site/freemarker/nationalindicatorlibrary/indicator.ftl]: Do you want GA event to be fired for this?

File: repository-data/webfiles/src/main/resources/site/freemarker/nationalindicatorlibrary/indicator.ftl [hunk 2]
```diff
@@ -115,19 +115,32 @@
...
+                                </li>
+                            </#if>
+
+                            <#if indicator.details.technicalSpecificationUrl?has_content>
+                                <li class="attachment">
+                                    <a href="${indicator.details.technicalSpecificationUrl}" target="_blank"><@fmt.message key="headers.technicalSpecification"/></a>
```
Comment (jacsamell) [inline repository-data/webfiles/src/main/resources/site/freemarker/nationalindicatorlibrary/indicator.ftl]: And GA for this as well?

File: site/src/main/java/uk/nhs/digital/nil/beans/Indicator.java [hunk 3]
```diff
@@ -68,6 +69,10 @@ public Calendar getAssuranceDate() {
         return getProperty(PropertyKeys.ASSURANCE_DATE);
     }
 
+    public boolean hasAttachments() {
+        return getAttachments() != null || !StringUtils.isEmpty(getDetails().getTechnicalSpecificationUrl()) || !StringUtils.isEmpty(getDetails().getQualityStatementUrl());
```
Comment (jacsamell) [inline site/src/main/java/uk/nhs/digital/nil/beans/Indicator.java]: Think you want `isEmpty(getAttachments())` as the prototype document will give you an empty list by default, not a null one.

File: repository-data/webfiles/src/main/resources/site/freemarker/nationalindicatorlibrary/indicator.ftl (hunk 1 above)
Comment (mchalejeiwc) [reply repository-data/webfiles/src/main/resources/site/freemarker/nationalindicatorlibrary/indicator.ftl]: I don't see why not. Will add for consistency.

**Pull request merged**